tools/update_rust_version.sh

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env bash
+
+
+# Ask rustup to pick the latest version that will work.
+# This requires rustup >= 1.20.0.
+echo "Updating rustc to latest compatible version..."
+rustup update nightly >/dev/null 2>/dev/null
+
+# Rerun the command so that it prints out the version it installed. We then have
+# to extract that from the output. If there is a better way to do this then we
+# should update this.
+RUSTUP_NIGHTLY_VERSION=`rustup update nightly 2>/dev/null`
+BEST_DATE=`echo $RUSTUP_NIGHTLY_VERSION | sed 's/.* \([0-9]*-[0-9]*-[0-9]*\).*/\1/g'`
```

Comment:
How is this the "best" date, as opposed to the latest date? Does `rustup update nightly` automatically picks a version compatible with all installed tools (rustfmt, clippy)? Or does it happily break them (since nightly is unstable anyway)?

Reply:
> Does rustup update nightly automatically picks a version compatible with all installed tools (rustfmt, clippy)?

Yep!